src/authorization.rs

```diff
@@ -21,15 +21,22 @@ pub trait Authenticatable {
     /// let mut client =
     ///     auth0_client::Auth0Client::new("client_id", "client_secret", "domain", "audience");
     ///
-    /// client.authenticate().await?;
+    /// client.authenticate(None).await?;
     /// # Ok(())
     /// # }
     /// ```
-    async fn authenticate(&mut self) -> Auth0Result<String>;
+    async fn authenticate(&mut self, credentials: Option<Credentials>) -> Auth0Result<String>;
```

Comment:
Can we have an enum instead of an option to make it more readable when not using credentials ?

Example:
```rust
enum AuthenticationMethod {
    Secrets,
    Credentials(Credentials)
}
```